producer-kafka/src/main/java/io/stargate/producer/kafka/configuration/DefaultConfigLoader.java

```diff
@@ -29,22 +30,32 @@
 import org.apache.kafka.clients.producer.ProducerConfig;
 
 public class DefaultConfigLoader implements ConfigLoader {
-
+  public static final String CONFIG_STORE_MODULE_NAME = "cdc.kafka";
   private static final Pattern CDC_KAFKA_PRODUCER_SETTING_PATTERN =
       Pattern.compile(String.format("^(%s)\\.(.*.)", CDC_KAFKA_PRODUCER_SETTING_PREFIX));
 
   @Override
-  public CDCKafkaConfig loadConfig(Map<String, Object> options) {
+  public CDCKafkaConfig loadConfig(ConfigStore configStore) {
+    return loadConfig(configStore.getConfigForModule(CONFIG_STORE_MODULE_NAME).getConfigMap());
+  }
+
+  private CDCKafkaConfig loadConfig(Map<String, Object> options) {
     String topicPrefixName = getTopicPrefixName(options);
     Map<String, Object> kafkaProducerSettings = filterKafkaProducerSettings(options);
     MetricsConfig metricsConfig = loadMetricsConfig(options);
     registerMetricsIfEnabled(kafkaProducerSettings, metricsConfig);
     String schemaRegistryUrl = getSchemaRegistryUrl(kafkaProducerSettings);
+    validateBootstrapServersPresent(kafkaProducerSettings);
     putProducerSerializersConfig(kafkaProducerSettings);
     return new CDCKafkaConfig(
         topicPrefixName, schemaRegistryUrl, kafkaProducerSettings, metricsConfig);
   }
 
+  private void validateBootstrapServersPresent(Map<String, Object> kafkaProducerSettings) {
+    // this should not throw
+    getStringSettingValue(kafkaProducerSettings, ProducerConfig.BOOTSTRAP_SERVERS_CONFIG);
```

Comment:
Would it be possible to provide methods like `getStringSettingValue` in the Config API module?.. either directly in hte config interface of by means of a utility class?

Reply:
That is actually an excellent idea. Created task:
https://github.com/stargate/stargate/issues/363
It does not block this PR - it can be addressed afterward, and then I will adapt this PR.